clients/graphql-kotlin-client/src/main/kotlin/com/expediagroup/graphql/client/extensions/AutomaticPersistedQueriesExtensions.kt

```diff
@@ -0,0 +1,19 @@
+package com.expediagroup.graphql.client.extensions
+
+import com.expediagroup.graphql.client.types.AutomaticPersistedQueriesExtension
+import com.expediagroup.graphql.client.types.GraphQLClientRequest
+import java.math.BigInteger
+import java.nio.charset.StandardCharsets
+import java.security.MessageDigest
+
+internal val MESSAGE_DIGEST: MessageDigest = MessageDigest.getInstance("SHA-256")
+
+fun GraphQLClientRequest<*>.getQueryId(): String =
+    String.format(
+        "%064x",
+        BigInteger(1, MESSAGE_DIGEST.digest(this.query.toByteArray(StandardCharsets.UTF_8)))
+    ).also {
+        MESSAGE_DIGEST.reset()
+    }
+
+fun AutomaticPersistedQueriesExtension.toQueryParamString() = """{"persistedQuery":{"version":$version,"sha256Hash":"$sha256Hash"}}"""
```

Comment:
whereas POST can also have query params , when using GraphQL with POST you have to pass query, operationName and extensions in the payload -- i would just call it `serialize()`.